infrastructure/code/code.go

```diff
@@ -116,10 +116,26 @@ func (sc *Scanner) SupportedCommands() []snyk.CommandName {
 }
 
 func (sc *Scanner) Scan(ctx context.Context, path string, folderPath string) (issues []snyk.Issue, err error) {
-	if !sc.isSastEnabled() {
+	sastResponse, err := sc.SnykApiClient.SastSettings()
+	method := "Scan"
+
+	if err != nil {
+		log.Error().Err(err).Str("method", method).Msg("couldn't get sast enablement")
+		sc.errorReporter.CaptureError(err)
+		return issues, errors.New("couldn't get sast enablement")
+	}
+
+	if !sc.isSastEnabled(sastResponse) {
```

Comment:
> it looks like we're calling the SnykApiClient.SastSettings() directly here 

Yes, because I need the `sastResponse` data for the new functionality I introduced. To avoid calling API twice, I requested `SnykApiClient.SastSettings()` once here and use its response for different functions.

> skipping the checks in /code/sast_enabled.go

The non-local-engine checks in `sast_enabled.go` are still in use. Line 128 is calling `isSastEnabled` function that has all non-local-engine checks there.

I modified and extracted the local-engine checks into `sast_local_engine.go` file.

Does this make sense?

Reply:
Makes sense, thank you for clarifying :)